src/components/structures/MatrixChat.js

```diff
@@ -1337,8 +1363,16 @@ export default React.createClass({
                 call: call,
             }, true);
         });
-        cli.on('Session.logged_out', function(call) {
+        cli.on('Session.logged_out', function(errObj) {
             if (Lifecycle.isLoggingOut()) return;
+
+            if (errObj.httpStatus === 401 && errObj.data && errObj.data['soft_logout']) {
+                console.warn("Soft logout issued by server - avoiding data deletion");
+                Lifecycle.softLogout();
+                dis.dispatch({actions: 'soft_logout'});
```

Comment:
`action` with an "s"? Also, this action doesn't seem to be listened to anywhere else?